socorro/tests/processor/test_cache_manager.py

```diff
@@ -0,0 +1,397 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import json
+import pathlib
+from unittest.mock import ANY
+
+from markus.testing import MetricsMock
+import pytest
+
+from socorro import settings
+from socorro.processor.cache_manager import (
+    count_sentry_scrub_error,
+    DiskCacheManager,
+    LastUpdatedOrderedDict,
+)
+
+
+class TestLastUpdatedOrderedDict:
+    def test_set(self):
+        lru = LastUpdatedOrderedDict()
+
+        lru["key1"] = 1
+        lru["key2"] = 2
+        assert list(lru.items()) == [("key1", 1), ("key2", 2)]
+
+        lru["key1"] = 3
+        assert list(lru.items()) == [("key2", 2), ("key1", 3)]
+
+    def test_touch(self):
+        lru = LastUpdatedOrderedDict()
+
+        lru["key1"] = 1
+        lru["key2"] = 2
+
+        lru.touch("key1")
+        assert list(lru.items()) == [("key2", 2), ("key1", 1)]
+
+    def test_popoldest(self):
+        lru = LastUpdatedOrderedDict()
+
+        lru["key1"] = 1
+        lru["key2"] = 2
+
+        oldest = lru.pop_oldest()
+        assert oldest == ("key1", 1)
+        assert list(lru.items()) == [("key2", 2)]
+
+
+@pytest.fixture
+def cm(tmp_path):
+    """Test cache manager setup with tmp_path."""
+    with settings.override(
+        SYMBOLS_CACHE_PATH=str(tmp_path),
+        SYMBOLS_CACHE_MAX_SIZE=10,
+    ):
+        cache_manager = DiskCacheManager()
```

Comment:
I thought about this a bunch. Should the `DiskCacheManager` class take these as arguments or should they be pulled in from settings? I decided to leave it as settings since we're testing the process which pulls from settings.